posts/airflow-ray.md

```diff
@@ -0,0 +1,285 @@
+        train_df, validation_df = data
+        evallist = [(validation_df, 'eval')]
+        evals_result = {}
+        config = {
+            "tree_method": "hist",
+            "eval_metric": ["logloss", "error"],
+        }
+        write("Start training")
+        bst = xgboost_ray.train(
+            params=config,
+            dtrain=train_df,
+            evals_result=evals_result,
+            ray_params=xgb.RayParams(max_actor_restarts=1, num_actors=8, cpus_per_actor=2),
+            num_boost_round=100,
+            evals=evallist)
+        return bst
+    ```
+
+3. Add the `@ray_task` decorator to each of these functions. 
+
+    ```python
+    @ray_task(**task_args)
+    def train_model(
+            data
+    ):
+        train_df, validation_df = data
+        evallist = [(validation_df, 'eval')]
+        evals_result = {}
+        config = {
+            "tree_method": "hist",
+            "eval_metric": ["logloss", "error"],
+        }
+        bst = xgb.train(
+            params=config,
+            dtrain=train_df,
+            evals_result=evals_result,
+            ray_params=xgb.RayParams(max_actor_restarts=1, num_actors=8, cpus_per_actor=2),
+            num_boost_round=100,
+            evals=evallist)
+        return bst
+    ```
+
+4. Create a DAG function with the logical flow of data
+
+    ```python
+    @dag(default_args=default_args, schedule_interval=None, start_date=days_ago(2), tags=['finished-modin-example'])
+    def task_flow_xgboost_modin():
+        build_raw_df = load_dataframe()
+        data = create_data(build_raw_df)
+        trained_model = train_model(data)
+
+    task_flow_xgboost_modin = task_flow_xgboost_modin()
+    ```
+
+5. Upload the DAG to Airflow, and your data pipeline is now a running Airflow DAG! 
+
+    With maybe 20 minutes of work, a data scientist can turn a local Python script into a high-scale, reproducible pipeline with the power of Ray distributed computing and Airflow orchestration. These data engineers can then take advantage of Airflow [variables](https://airflow.apache.org/docs/apache-airflow/stable/concepts.html#variables), [connections](https://airflow.apache.org/docs/apache-airflow/stable/concepts.html#connections), scheduling intervals, and many other features that have made Airflow so ubiquitous with data teams requiring production-grade scheduling and orchestration.
+
+    A basic Ray workflow might now appear together with Airflow like this:
+    ![airflow-ui-ray-tune-xgboost](https://user-images.githubusercontent.com/307956/117888159-93bd5800-b266-11eb-84d0-b0acc7cb2d59.png)
+
+### Passing data between tasks: faster with Plasma
+
+Experienced Airflow users will notice in the example above we appear to be passing entire dataframes between tasks without explicitly sending those data chunks to external storage. With traditional XCom, this would be essentially impossible because Airflow stores each piece of data sent between tasks in a single cell of the metadata DB. 
```

Comment:
```suggestion
In the example above, Airflow users might notice that we appear to be passing entire dataframes between tasks without explicitly sending those data chunks to external storage. With traditional XCom, this would be essentially impossible because Airflow stores each piece of data sent between tasks in a single cell of the metadata DB. 
```